src/block_disagg/block_disagg_write.c

```diff
@@ -110,7 +113,14 @@ __wt_block_disagg_write_internal(WT_SESSION_IMPL *session, WT_BLOCK_DISAGG *bloc
     page_id = block_meta->page_id;
 
     /* Get the checkpoint ID. */
-    checkpoint_id = 1; /* XXX Use the current checkpoint order instead of hardcoding this. */
+    WT_ACQUIRE_READ(checkpoint_id, conn->disaggregated_storage.global_checkpoint_id);
+
+    /* Check that the checkpoint ID matches the current checkpoint in the page log. */
+    if (block_disagg->plhandle->page_log->pl_get_open_checkpoint != NULL) {
+        WT_RET(block_disagg->plhandle->page_log->pl_get_open_checkpoint(
+          block_disagg->plhandle->page_log, &session->iface, &page_log_checkpoint_id));
+        WT_ASSERT(session, checkpoint_id == page_log_checkpoint_id);
```

Comment:
I think we have a WT_ASSERT_ALWAYS, that should be used here, so the check is even in release builds.  BTW, do we expect PALI's get_open_checkpoint to hit the network?  I suppose it could cache the value.

Reply:
Makes sense—I switched it to WT_ASSERT_ALWAYS. I expect that `get_open_checkpoint` would not need to hit the network, but just check the state on the server.